src/type.zig

```diff
@@ -1640,6 +1640,7 @@ pub const Type = extern union {
             .Int,
             .Float,
             .ErrorSet,
+            .ErrorUnion,
```

Comment:
This should check if the error union payload is also a valid variable type, but I'll just fix that when I also fix the incorrect cast you pointed out.